[Decomposition] baddbmm

salilsdesai opened the pull request:
Summary:
Moving decomposition of baddbmm from _inductor/decomposition.py and include it in core_aten_decompositions

https://github.com/pytorch/pytorch/blob/ff38c0e2f9cae35378553c38ccf7188007fed938/torch/_inductor/decomposition.py#L203

Test Plan: Phabricator + OSS Tests

Differential Revision: D48871741




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

SherlockNoMad commented on torch/_decomp/decompositions.py:
```diff
@@ -4029,6 +4030,21 @@ def inplace_op(*args, **kwargs):
 
 
+@register_decomposition([aten.baddbmm])
+def baddbmm(self, batch1, batch2, beta=1, alpha=1):
```
add 
```
@out_wrapper()
@pw_cast_for_opmath
```

SherlockNoMad reviewed (approved):
per comment.

salilsdesai committed:
[Decomposition] baddbmm (#108534)

Summary:
bypass-github-pytorch-ci-checks

Pull Request resolved: https://github.com/pytorch/pytorch/pull/108534

Moving decomposition of baddbmm from _inductor/decomposition.py and include it in core_aten_decompositions

https://github.com/pytorch/pytorch/blob/ff38c0e2f9cae35378553c38ccf7188007fed938/torch/_inductor/decomposition.py#L203

Test Plan: Phabricator + OSS Tests

Reviewed By: SS-JIA

Differential Revision: D48871741

fbshipit-source-id: d7e2921112a0b333a389693cd45f33742a3f5b4c

peterbell10 commented on torch/_decomp/decompositions.py:
```diff
@@ -4034,6 +4035,23 @@ def inplace_op(*args, **kwargs):
 
+@out_wrapper()
+@pw_cast_for_opmath
+@register_decomposition([aten.baddbmm])
```
You're registering the decomposition before applying the `pw_cast_for_opmath` wrapper. This PR shouldn't have been merged with the tests still skipped.
